nncf/quantization/algorithms/min_max/algorithm.py

```diff
@@ -175,32 +219,54 @@ def _get_stat_collector(self, quantizer_config: QuantizerConfig) -> TensorStatis
+        """
+        Returns default quantizer configuration, based on the provided constraints.
+
+        :param constraints: Quantization constraints.
+        :return: Quantizer config.
+        """
+        qconfig = deepcopy(self._parameters.DEFAULT_QCONFIG)
+        if constraints is not None:
+            qconfig = constraints.apply_constraints_to(qconfig)
+        return qconfig
+
     def _get_quantizer_setup(self, nncf_graph: NNCFGraph) -> SingleConfigQuantizerSetup:
         """
         Returns SingleConfigQuantizerSetup instance based on the input NNCFGraph.
 
         :param nncf_graph: NNCFGraph instance.
         :return: SingleConfigQuantizerSetup for the current NNCFGraph entity.
         """
-        ip_graph = InsertionPointGraph(nncf_graph)
-        weight_nodes = nncf_graph.get_nodes_by_metatypes(self._backend_entity.layers_with_weights_metatypes)
-        quantizable_layer_nodes = [QuantizableWeightedLayerNode(weight_node, [QuantizerConfig()])
-                                   for weight_node in weight_nodes]
-        pattern = self._backend_entity.hw_fused_patterns.get_full_pattern_graph()
-        ip_graph = ip_graph.get_ip_graph_with_merged_hw_optimized_operations(pattern, quantizable_layer_nodes)
-
         hw_config_type = self._parameters.target_device
         hw_config_path = self._backend_entity.hw_config.get_path_to_hw_config(hw_config_type)
         hw_config = self._backend_entity.hw_config.from_json(hw_config_path)
-        default_config = deepcopy(self._parameters.DEFAULT_QCONFIG)
-
+        weight_nodes = nncf_graph.get_nodes_by_metatypes(self._backend_entity.layers_with_weights_metatypes)
+        default_weight_qconfig = self._get_default_qconfig(
+            self._parameters.global_quantizer_constraints[QuantizerGroup.WEIGHTS])
+        weighted_node_and_qconf_lists = assign_qconfig_lists_to_modules(nodes_with_weights=weight_nodes,
+                                                                        default_weight_qconfig=default_weight_qconfig,
+                                                                        global_weight_constraints=
+                                                                        self._parameters.global_quantizer_constraints[
+                                                                            QuantizerGroup.WEIGHTS],
+                                                                        scope_overrides_dict=None,
+                                                                        hw_config=hw_config)
+        quantizable_layer_nodes = [QuantizableWeightedLayerNode(node, qconf_list) for node, qconf_list
+                                   in weighted_node_and_qconf_lists.items()]
+        pattern = self._backend_entity.hw_fused_patterns.get_full_pattern_graph()
+        ip_graph = InsertionPointGraph(nncf_graph)
+        ip_graph = ip_graph.get_ip_graph_with_merged_hw_optimized_operations(pattern, quantizable_layer_nodes)
```

Comment:
This name is also overkill: `get_ip_graph_with_merged_hw_optimized_operations`. It can be reduced to `ip_graph.fuse_patterns(...)`. Please avoid this in the future.

Reply:
This is the common method worked on IPGraph. I jam just reusing it

Reply:
ok. I just highlighted that this is not a good example of naming.